tests/test_fips.py

```diff
@@ -237,3 +270,95 @@ def test_gnutls_binary(container_per_test: ContainerData) -> None:
+        "sha512_224": "8b38cae68bdc4ef3336f61f68fa1b3a373abb330033a59fc51c31647f5b622b94204aa999acc71edc1f80ba7542abcdce28213a8925815a3",
+        "sha512_256": "04559e6cddea9a198f69bbd2a4c882d2fa3b9cda0edf18eb1582df42d88f7dba198bc813cc1b1bc065dba1bf261c1c9e3c9e92eb66c25638a447a744c38690ba",
+        "sha3-224": "7639f4678682e8c6b0968c894b9b58cac8cfbb6b44e0b7e99126229cdc0d4419c6e5dbc2e5fbb9fcb9dc6ea16d6b3cbca34aabb618e2f782",
+        "sha3-256": "9cd04d17df1c852d2b13536105c400e26d46ee0e865bd6d91d8683a2979fcb59265a271f568a62eb8c64e5cbedbdfd41d996303de25868af9b1892bda0bbcdfa",
+        "sha3-384": "b8b1ec43880fc0495a520434e379dce51fb0959f244c70b75555cae44dbac99f231bed131ee5a01ab8cd6e66e5da9b68ed7dc3dd68ec7a5898986cde448f567512a0f2cef69d1ed4e5fa2033950642c64f3a762b59576df4117cca5e7a52c188",
+        "sha3-512": "0bc583f093fb557e5aa6c3020cc93e9065a01600f3c8e1be17c5026799e4434c701a423ade5edae77491b7ee05c78cdc6e01af22bdc67fcd351fb1ab9a1936abc9a2f69add89697a8b935ce1bbc3d5c21f276b9e2377015003f98543bceffb8a2e1ec8d96fc99bd39d88ca6a0fd69d812362eda0937de4e3b3f0d94093a5b5cc",
+    }
+
+    unsupported_fips_gcrypt_digests_15_3 = ["sha512_224", "sha512_256"]
+    unsupported_non_fips_gcrypt_digests_15_3 = ["sm3"]
+
+    for digest in FIPS_GCRYPT_DIGESTS:
+        res = container_per_test.connection.run_expect(
+            [0, 1], f"/bin/fips-test-gcrypt {digest}"
+        )
+
+        if (
+            OS_VERSION == "15.3"
+            and digest in unsupported_fips_gcrypt_digests_15_3
+        ):
+            assert (
+                res.rc == 1
+                and f"Unknown message digest {digest}" in res.stderr
+            ), f"Hash calculation unexpectedly supported for {digest} in os version {OS_VERSION}"
+            continue
+
+        assert (
+            res.rc == 0
+            and "Digest is: " + expected_fips_gcrypt_digests[digest]
+            in res.stdout
+        )
+
+    for digest in NONFIPS_GCRYPT_DIGESTS:
+        non_fips_call = container_per_test.connection.run_expect(
+            [0, 1], f"/bin/fips-test-gcrypt {digest}"
+        )
+        if non_fips_call.rc != 1 and non_fips_call.stdout.strip().startswith(
+            "Digest is: "
+        ):
+            if OS_VERSION == "15.6" and digest == "md5":
+                pytest.xfail(
+                    "libgcrypt successfully calculates md5 digest in FIPS mode, bsc#1229903"
+                )
+
+            pytest.xfail(
+                "libgcrypt mistakenly calculates message digests for Non FIPS algorithms, bsc#1229856"
+            )
+
+        if (
+            OS_VERSION == "15.3"
+            and digest in unsupported_non_fips_gcrypt_digests_15_3
+        ):
+            assert (
+                non_fips_call.rc == 1
+                and f"Unknown message digest {digest}" in non_fips_call.stderr
+            ), f"Hash calculation unexpectedly supported for {digest} in os version {OS_VERSION}"
+            continue
```

Comment:
Imho it would be better to not call `/bin/fips-test-grypt` on the unsupported hashes.